jobs/services/vaultwarden.hcl

```diff
@@ -0,0 +1,67 @@
+job "vaultwarden" {
+  datacenters = ["aperture"]
+  type        = "service"
+
+  group "vaultwarden" {
+    count = 1
+
+    network {
+      port "http" {
+        to = 80
+      }
+    }
+
+    service {
+      name = "vaultwarden"
+      port = "http"
+
+      tags = [
+        "traefik.enable=true",
+        "traefik.http.routers.vaultwarden.rule=Host(`vault.redbrick.dcu.ie`)",
+        "traefik.http.routers.vaultwarden.entrypoints=websecure",
+        "traefik.http.routers.vaultwarden.tls.certresolver=lets-encrypt",
+      ]
+    }
+
+    task "vaultwarden" {
+      driver = "docker"
+
+      config {
+        image = "vaultwarden/server:latest-alpine"
+        ports = ["http"]
+
+        volumes = [
+          "/storage/nomad/vaultwarden:/data"
+        ]
+      }
+
+      template {
+        data = <<EOF
+DOMAIN=https://vault.redbrick.dcu.ie
+DATABASE_URL=postgresql://{{ key "vaultwarden/db/user" }}:{{ key "vaultwarden/db/password" }}@postgres.service.consul:5432/{{ key "vaultwarden/db/name" }}
+SIGNUPS_ALLOWED=false
+INVITATIONS_ALLOWED=true
+# This is not the actual token, but a hash of it. Vaultwarden does not like the actual token.
+ADMIN_TOKEN={{ key "vaultwarden/admin/hash" }}
+EOF
+
+        destination = "local/env"
+        env         = true
+      }
+#YUBICO_CLIENT_ID={{ key "vaultwarden/yubico/client_id" }}
+#YUBICO_SECRET_KEY={{ key "vaultwarden/yubico/secret_key" }}
```

Comment:
Not entirely sure, though it seems like they will be needed if we want to allow users to use yubikey otp.

See: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-Yubikey-OTP-authentication

Reply:
I suppose it's not necessary since yubikeys support U2F: https://github.com/dani-garcia/vaultwarden/wiki/Enabling-U2F-%28and-FIDO2-WebAuthn%29-authentication


Will remove these lines now

Reply:
https://vaultwarden.discourse.group/t/multiple-yubikeys-howto-configure-environment-variables/250/4

Also relevant information

Reply:
Leave them there for now, but add a comment with the above link. It should explain to future admins why it's there